cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectorWatcher.java

```diff
@@ -0,0 +1,163 @@
+                                                        kafkaConnector.getMetadata().getName(), action == Action.DELETED ? null : kafkaConnector)
+                                                        .compose(reconcileResult -> {
+                                                            LOGGER.infoCr(reconciliation, "reconciled");
+                                                            return Future.succeededFuture(reconcileResult);
+                                                        }));
+                                    }
+                                }
+                            });
+                } else {
+                    updateStatus(new Reconciliation("connector-watch", connectOperator.kind(),
+                                    kafkaConnector.getMetadata().getNamespace(), null),
+                            new InvalidResourceException("Resource lacks label '"
+                                    + Labels.STRIMZI_CLUSTER_LABEL
+                                    + "': No connect cluster in which to create this connector."),
+                            kafkaConnector, connectOperator.connectorOperator);
+                }
+
+                break;
+            case ERROR:
+                LOGGER.errorCr(new Reconciliation("connector-watch", connectorKind, connectName, connectorNamespace), "Failed {} {} in namespace {} ", connectorKind, connectorName, connectorNamespace);
+                break;
+            default:
+                LOGGER.errorCr(new Reconciliation("connector-watch", connectorKind, connectName, connectorNamespace), "Unknown action: {} {} in namespace {}", connectorKind, connectorName, connectorNamespace);
+        }
+    }
+
+    @Override
+    public void onClose(WatcherException e) {
+        onClose.accept(e);
+    }
+
+    /**
+     * Update the status of the Kafka Connector custom resource
+     *
+     * @param reconciliation        Reconciliation marker
+     * @param error                 Throwable indicating if any errors occurred during the reconciliation
+     * @param kafkaConnector2       Latest version of the KafkaConnector resource where the status should be updated
+     * @param connectorOperations   The KafkaConnector operations for updating the status
+     */
+    public static void updateStatus(Reconciliation reconciliation, Throwable error, KafkaConnector kafkaConnector2, CrdOperator<?, KafkaConnector, ?> connectorOperations) {
+        KafkaConnectorStatus status = new KafkaConnectorStatus();
+        StatusUtils.setStatusConditionAndObservedGeneration(kafkaConnector2, status, error);
+        StatusDiff diff = new StatusDiff(kafkaConnector2.getStatus(), status);
+        if (!diff.isEmpty()) {
+            KafkaConnector copy = new KafkaConnectorBuilder(kafkaConnector2).build();
+            copy.setStatus(status);
+            connectorOperations.updateStatusAsync(reconciliation, copy);
+        }
+    }
```

Comment:
This seems a bit weird. You seem to have the same method in the `connectOperator`. Why don't you just call it there? You seem to duplicate it.